lib/auth/native/native.go

```diff
@@ -42,16 +52,46 @@ type keyPair struct {
 type nauth struct {
 	generatedKeysC chan keyPair
 	closeC         chan bool
-	closed         int32
+	mutex          sync.Mutex
+	// in production set to nil (used for tests)
+	testKey *keyPair
 }
 
+// New returns a pointer to a key generator for production purposes
 func New() *nauth {
-	n := nauth{
-		generatedKeysC: make(chan keyPair, PrecalculatedKeysNum),
-		closeC:         make(chan bool),
+	singleton.mutex.Lock()
+	defer singleton.mutex.Unlock()
+
+	if singleton.generatedKeysC == nil && PrecalculatedKeysNum > 0 {
+		singleton.generatedKeysC = make(chan keyPair, PrecalculatedKeysNum)
+		go singleton.precalculateKeys()
+	}
+	return &singleton
+}
+
+// NewTestInstance
+func SetTestKeys() {
```

Comment:
I find it a bit scary to bake test code into production flow. How about instead of setting property of `nauth` we will do the following:

* singleton variable `Authority` will be part of `lib/auth` package
* by default it will return initialized native authority just as your code above
* it will have a global function just as the one you wrote that can be called by tests to override global vairable `SetAuthority(a Authority)` that will be used to pass `testauthority` instance to all users on `init` everywhere in tests

This way we will keep hardcoded key out of the `native` authority flow and still have the same effect